drivers/ethernet/eth_nxp_enet.c

```diff
@@ -157,105 +145,85 @@ static void net_if_mcast_cb(struct net_if *iface,
 		ENET_LeaveMulticastGroup(config->base, mac_addr.addr);
 	}
 }
-#endif /* CONFIG_NET_NATIVE_IPV4 || CONFIG_NET_NATIVE_IPV6 */
-
-/*
- *****************
- * PTP Functions *
- *****************
- */
 
 #if defined(CONFIG_PTP_CLOCK_NXP_ENET)
 static bool eth_get_ptp_data(struct net_if *iface, struct net_pkt *pkt)
 {
-	int eth_hlen;
+	struct net_eth_vlan_hdr *hdr_vlan = (struct net_eth_vlan_hdr *)NET_ETH_HDR(pkt);
+	struct ethernet_context *eth_ctx = net_if_l2_data(iface);
+	bool pkt_is_ptp;
 
-#if defined(CONFIG_NET_VLAN)
-	struct net_eth_vlan_hdr *hdr_vlan;
-	struct ethernet_context *eth_ctx;
-	bool vlan_enabled = false;
-
-	eth_ctx = net_if_l2_data(iface);
 	if (net_eth_is_vlan_enabled(eth_ctx, iface)) {
-		hdr_vlan = (struct net_eth_vlan_hdr *)NET_ETH_HDR(pkt);
-		vlan_enabled = true;
-
-		if (ntohs(hdr_vlan->type) != NET_ETH_PTYPE_PTP) {
-			return false;
-		}
-
-		eth_hlen = sizeof(struct net_eth_vlan_hdr);
-	} else
-#endif /* CONFIG_NET_VLAN */
-	{
-		if (ntohs(NET_ETH_HDR(pkt)->type) != NET_ETH_PTYPE_PTP) {
-			return false;
-		}
-
-		eth_hlen = sizeof(struct net_eth_hdr);
+		pkt_is_ptp = ntohs(hdr_vlan->type) == NET_ETH_PTYPE_PTP;
+	} else {
+		pkt_is_ptp = ntohs(NET_ETH_HDR(pkt)->type) == NET_ETH_PTYPE_PTP;
 	}
 
-	net_pkt_set_priority(pkt, NET_PRIORITY_CA);
+	if (pkt_is_ptp) {
+		net_pkt_set_priority(pkt, NET_PRIORITY_CA);
+	}
 
-	return true;
+	return pkt_is_ptp;
 }
 
 
-#if defined(CONFIG_NET_L2_PTP)
```

Comment:
Why is this deleted?

Reply:
CONFIG_PTP_CLOCK_NXP_ENET depends on CONFIG_NET_L2_PTP anyways